base/src/com/google/idea/blaze/base/sync/workspace/ExecutionRootPathResolver.java

```diff
@@ -73,49 +78,55 @@ public File resolveExecutionRootPath(ExecutionRootPath path) {
     if (path.isAbsolute()) {
       return path.getAbsoluteOrRelativeFile();
     }
-    if (isInWorkspace(path)) {
-      return workspacePathResolver.resolveToFile(path.getAbsoluteOrRelativeFile().getPath());
+    String firstPathComponent = getFirstPathComponent(path.getAbsoluteOrRelativeFile().getPath());
+    if (buildArtifactDirectories.contains(firstPathComponent)) {
+      // Build artifacts accumulate under the execution root, independent of symlink settings
+      return path.getFileRootedAt(executionRoot);
+    }
+    if (firstPathComponent.equals("external")) { // In external workspace
```

Comment:
Hey hey! Thanks for taking another look, @tpasternak.

I'd think we'd need to modify execution root paths--and not just directories--because one might try to browse to an external file path. More generally, those execroot paths are unstable for all files, not just directories, so I'd think we'd always want to point to the stable paths, lest they disappear under the editor when a build gets run that uses different external workspaces. This ended up being important for Xcode/VSCode, anyway. I think this is a case we want to defend against. (Unless I'm misunderstanding your question; you all are the experts, immersed in this day-to-day.)